src/components/ContactsList/ContactsList.jsx

```diff
@@ -1,66 +1,117 @@
+    return _.find(contacts, c => c.id === id)
+  }
+  render() {
+    const { displayImportation, onSelect, selection, groups } = this.props
+    const { displayedContactId } = this.state
+    return (
+      <Query query={query}>
+        {({ data: contacts, fetchStatus }) => {
+          if (fetchStatus === 'loading') {
+            return <Spinner size="xxlarge" />
+          }
+          if (contacts.length === 0) {
+            return <ContactsEmptyList displayImportation={displayImportation} />
+          } else {
+            const sortedContacts = [...contacts].sort(sortLastNameFirst)
+            const categorizedContacts = sortedContacts.reduce(
+              (acc, contact) => {
+                const name = buildLastNameFirst(contact)
+                const header = name[0] || 'EMPTY'
+                acc[header] = acc[header] || []
+                acc[header].push(contact)
+                return acc
+              },
+              {}
+            )
+            return (
+              <div className="list-wrapper">
+                <ol className="list-contact">
+                  {Object.keys(categorizedContacts).map(header => (
+                    <li key={`cat-${header}`}>
+                      <ContactHeaderRow key={header} header={header} />
+                      <ol className="sublist-contact">
+                        {categorizedContacts[header].map(contact => (
+                          <li key={`contact-${contact._id}`}>
+                            <ContactRow
+                              key={contact._id}
+                              contact={contact}
+                              groups={groups}
+                              selection={
+                                onSelect && {
+                                  onSelect: () => {
+                                    onSelect(contact)
+                                  },
+                                  selected:
+                                    _.find(
+                                      selection,
+                                      s => s.id === contact._id
+                                    ) !== undefined
+                                }
+                              }
+                              onClick={e => this.onClick(e, contact._id)}
+                            />
+                          </li>
+                        ))}
+                      </ol>
+                    </li>
+                  ))}
+                </ol>
+                <div>
+                  {displayedContactId && (
+                    <ContactCardModal
+                      onClose={this.hideContactCard}
+                      contact={this.getContactById(
+                        contacts,
+                        displayedContactId
+                      )}
+                      groups={groups}
+                    />
+                  )}
+                </div>
+              </div>
+            )
+          }
+        }}
+      </Query>
```

Comment:
Not blocking for this PR, but I think we should split this component next chance we get — separating the loading state, the modal management, etc.